src/lib/functions/registry.mjs

```diff
@@ -145,11 +148,12 @@ export class FunctionsRegistry {
       )
     }
 
-    // This fixes the bug described here https://github.com/netlify/zip-it-and-ship-it/issues/637
-    // If the current function's file is a zip bundle, we ignore it and log a helpful message.
+    // If the function file is a ZIP, we extract it and rewire its main file to
```

Comment:
We need to do this so that the functions have all its dependencies available, otherwise we'd need to load modules from `.netlify/plugins/node_modules`, which isn't part of the module resolution path. We know we've already bundled the functions as part of the build process, so we can extract their ZIP files and use the exact files that will be used in Lambda, in the spirit of simulation the production environment as closely as possible.